File: PDOO/Napakalaki-Ruby/lib/numeric_bad_consequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module NapakalakiGame
  class NumericBadConsequence < BadConsequence

    def initialize (aText, someLevels, someVisibleTreasures, someHiddenTreasures)
      super(aText, someLevels)
      @nVisibleTreasures = someVisibleTreasures
      @nHiddenTreasures = someHiddenTreasures
    end

    def isEmpty
      resultado=false

      if(@nVisibleTreasures == 0 && @nHiddenTreasures == 0)
        resultado=true
      end

      return resultado

    end

    def getNVisibleTreasures
      @nVisibleTreasures
    end

    def getNHiddenTreasures
      @nHiddenTreasures
    end
    
    def getSpecificVisibleTreasures
      return Array.new
    end
  
    def getSpecificHiddenTreasures
      return Array.new
    end
    
    def getDeath
      return false
    end

    def substractVisibleTreasure(t)
      @nVisibleTreasures = @nVisibleTreasures - 1

      if(@nVisibleTreasures < 0)
        @nVisibleTreasures = 0
      end

    end

    def substractHiddenTreasure(t)
      @nHiddenTreasures = @nHiddenTreasures - 1
      if (@nHiddenTreasures < 0)
        @nHiddenTreasures = 0
      end
    end

    def adjustToFitTreasureLists(v, h)
      if(@nVisibleTreasures >= v.size)
        nV = v.size
      else
        nV = @nVisibleTreasures
      end
      if(@nHiddenTreasures >= h.size)
        nH = h.size
      else
        nH = @nHiddenTreasures
      end
      bc = NumericBadConsequence.new(@text, 0, nV, nH)
      return bc
    end
    
    def to_s
        return "#{super} Visible Treasures: #{@nVisibleTreasures} Hidden Treasures: #{@nHiddenTreasures} Death: #{@death}" 
    end

  end

end

File: PDOO/Napakalaki/src/NapakalakiGame/NumericBadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 * @author nestor
 */
public class NumericBadConsequence extends BadConsequence {

    private int nVisibleTreasures = 0;
    private int nHiddenTreasures = 0;

    public NumericBadConsequence(String t, int l, int nVisible, int nHidden) {
        super(t, l);
        this.nVisibleTreasures = nVisible;
        this.nHiddenTreasures = nHidden;
    }

    public boolean isEmpty() {
        boolean resultado = false;

        if (nVisibleTreasures == 0 && nHiddenTreasures == 0) {
            resultado = true;
        }

        return resultado;
    }

    public int getNVisibleTreasures() {
        return nVisibleTreasures;
    }

    public int getNHiddenTreasures() {
        return nHiddenTreasures;
    }
    
    public ArrayList<TreasureKind> getSpecificVisibleTreasures() {
        return new ArrayList();
    }

    public ArrayList<TreasureKind> getSpecificHiddenTreasures() {
        return new ArrayList();
    }
    
    public boolean getDeath() {
        return false;
    }

    public void substractVisibleTreasure(Treasure t) {
        nVisibleTreasures--;
        if (nVisibleTreasures < 0) {
            nVisibleTreasures = 0;
        }
    }

    public void substractHiddenTreasure(Treasure t) {
        nHiddenTreasures--;
        if (nHiddenTreasures < 0) {
            nHiddenTreasures = 0;
        }
    }

    public BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v, ArrayList<Treasure> h) {
        int nV = 0, nH = 0;

        if (this.nVisibleTreasures >= v.size()) {
            nV = v.size();
        } else {
            nV = this.nVisibleTreasures;
        }
        if (this.nHiddenTreasures >= h.size()) {
            nH = h.size();
        } else {
            nH = this.nHiddenTreasures;
        }
        return new NumericBadConsequence(this.getText(), 0, nV, nH);
    }

    public String toString() {
        return super.toString() + " Visible Treasures = " + Integer.toString(nVisibleTreasures)
                + " Hidden Treasures = " + Integer.toString(nHiddenTreasures) + "\n";
    }
}
